.gitignore

```diff
@@ -11,6 +11,10 @@
 !/bin
 /bin/*
 !/bin/brew
+!/share
+/share/*
+!/share/doc
+/share/doc/*
```

Comment:
Try to make a change to something in `share/doc/homebrew` and `git add` it and see if it needs `-f`. If it doesn't: could just me my setup that's weird.